src/worker.js

```diff
@@ -274,3 +264,57 @@ this.onmessage = function(e) {
     throw e;
   }
 };
+
+#if ENVIRONMENT_MAY_BE_NODE
+// Node.js support
+if (typeof process === 'object' && typeof process.versions === 'object' && typeof process.versions.node === 'string') {
+  // Create as web-worker-like an environment as we can.
+  self = {
+    location: {
+      href: __filename
+    }
+  };
+
+  var onmessage = this.onmessage;
+
+  var nodeWorkerThreads = require('worker_threads');
+
+  Worker = nodeWorkerThreads.Worker;
+
+  var parentPort = nodeWorkerThreads.parentPort;
+
+  parentPort.ref(); // TODO: investigate when we need this
```

Comment:
As long as the `message` listener two lines below is present, this should never really be needed? We manage ref’ed-ness through the listener count by default.

Reply:
Thanks, I wasn't sure how that all works. We'll always set up the listener, so I'll remove the `ref()` calls.